app/components/progress_badge_component.rb

```diff
@@ -19,11 +19,17 @@ def circumference
   end
 
   def badge
-    course_badges.fetch(course.title, 'icons/odin-icon.svg')
+    @course.badge_uri || fallback_icon
   end
 
   def borderless_badge
-    borderless_badges.fetch(course.title, 'icons/odin-icon.svg')
```

Comment:
Mixed thoughts about this approach, although I do think it is an ugprade from doing it in the hash

Reply:
Definitely an upgrade, nicely done!